### PR TITLE
Use sleef on macOS Apple silicon by default

### PR DESCRIPTION
**Pull request by helloguo**
Use sleef ~~for aarch64~~ on macOS Apple silicon by default.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by helloguo** [comment]
Test on MacBook Air M2 with Llama-2-7b model using AOT Inductor. Here are the profiling:

Before: `Vectorized<float>::exp()` calls `expf`, which is from libsystem_m.dylib.

sleef_before

After: `Vectorized<float>::exp()` calls `Sleef_expf4_u10`.

sleef_after

**Review by digantdesai — APPROVED** [review]
Looks good to me.

File: CMakeLists.txt [hunk 1]
```diff
@@ -276,7 +276,7 @@ if(NOT DEFINED USE_VULKAN)
       "ANDROID" OFF)
 endif()
 
-option(USE_SLEEF_FOR_ARM_VEC256 "Use sleef for arm" OFF)
+option(USE_SLEEF_FOR_ARM_VEC256 "Use sleef for arm" ON)
```
Comment (malfet) [inline CMakeLists.txt:279]: Shouldn't this be a conditional option on ARM cpu?

**Review by malfet — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (helloguo) [reply CMakeLists.txt:279]: Good question. Maybe it's better to leave USE_SLEEF_FOR_ARM_VEC256 default off, and use another flag for Apple silicon. I will make some change accordingly.

Comment (helloguo) [reply CMakeLists.txt]: Hi @malfet I made some change to enable sleef on macOS with Apple silicon only, other ARM cpus will not be impacted. Do you think it's ok to merge? If so, can you merge it? I don't have the permission. There are still several test failures on linux, which seem unrelated to this change.

**Pull request closed without merging**